### PR TITLE
[doc] How to add interrupts to IP

### PR DESCRIPTION
**Pull request by eunchan**
This commit adds a section in Getting Started HW.
The section describes how-to add/revise the interrupts.
Currently, the designer needs to revise many parts (DV, SW, unittest)
when updating the interrupts.

### CODE REVIEW

**Comment by eunchan** [comment]
Changing interrupts in an IP is such a pain... Need document to describe the step. :)

File: doc/ug/getting_started_hw_design.md
Comment (timothytrippel) [inline doc/ug/getting_started_hw_design.md:118]: Could we just update the [`hw/Makefile` ](https://github.com/lowRISC/opentitan/blob/master/hw/Makefile) to call this too? Then we don't need this extra step.

File: doc/ug/getting_started_hw_design.md
Comment (timothytrippel) [inline doc/ug/getting_started_hw_design.md:124]: nit: "Top Plic" --> "Top PLIC"?

File: doc/ug/getting_started_hw_design.md
Comment (timothytrippel) [inline doc/ug/getting_started_hw_design.md:130]: (Not an actionable item, just commenting to document this / through this thought out there): would be nice (in the future) if we just parameterized this whole DIF/unittest combo and autogened the whole thing from the HJSONs.

**Review by timothytrippel — APPROVED** [review]
LGTM, modulo a couple comments. Thanks @eunchan !

File: doc/ug/getting_started_hw_design.md
Comment (tjaychen) [reply doc/ug/getting_started_hw_design.md:118]: yeah i tend to agree...
@timothytrippel what do you think?

Comment (tjaychen) [reply doc/ug/getting_started_hw_design.md:118]: oops, sorry i thought @eunchan left this comment haha.
But I totally agree we should do this.

File: doc/ug/getting_started_hw_design.md
Comment (rswarbrick) [inline doc/ug/getting_started_hw_design.md:128]:
```suggestion
If the number of interrupts has changed, you need to revise the SW unit test manually.
```

File: doc/ug/getting_started_hw_design.md
Comment (rswarbrick) [reply doc/ug/getting_started_hw_design.md:130]: Hear, hear!

File: doc/ug/getting_started_hw_design.md
Comment (eunchan) [reply doc/ug/getting_started_hw_design.md:121]: Google has its own style guide for clang-format mandated. Is there a simple way to avoid it?

Comment (timothytrippel) [reply doc/ug/getting_started_hw_design.md:121]: Are you referring to the fact that `util/make_new_dif.py` invokes clang-format and by calling `util/make_new_dif.py` in the `hw/Makefile` we will a similar problem that was addressed in #8909? If so, I have an idea how we can avoid the clang-format issues, but we should probable address that in another PR. So maybe for now, let's merge this the way it is, and once merged, I will file a separate issue about invoking `util/make_new_dif.py` from  `hw/Makefile`, and then I can implement the PR to address this and fix the clang-format issues. Does that sound good?

File: doc/ug/getting_started_hw_design.md
Comment (rswarbrick) [inline doc/ug/getting_started_hw_design.md:111]: Grammar nit: I think you want either "with the above command" or (better) "with the command above" (missing "the")
```suggestion
After you revise the IP `.hjson`, IP top module, IP register interface, and DV testbench `tb.sv`, you can re-generate top-levels with the command above.
```

File: doc/ug/getting_started_hw_design.md
Comment (rswarbrick) [inline doc/ug/getting_started_hw_design.md:117]: Grammar nit:
```suggestion
If the interrupt name is changed or a new interrupt is introduced, the DIF for the IP should be re-generated.
```

File: doc/ug/getting_started_hw_design.md
Comment (rswarbrick) [reply doc/ug/getting_started_hw_design.md:125]: This still needs fixing.

**Review by rswarbrick — APPROVED** [review]

**Comment by eunchan** [comment]
Merging now.
image

File: doc/ug/getting_started_hw_design.md
Comment (timothytrippel) [reply doc/ug/getting_started_hw_design.md]: Captured in #9650.

**Pull request merged**
